server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java

```diff
@@ -330,10 +342,42 @@ public void run() {
     public void testOverFlow() throws Exception {
         long fromSeqNo = randomLongBetween(0, 5);
         long toSeqNo = randomLongBetween(Long.MAX_VALUE - 5, Long.MAX_VALUE);
-        try (Translog.Snapshot snapshot = engine.newChangesSnapshot("test", fromSeqNo, toSeqNo, true, randomBoolean())) {
+        try (Translog.Snapshot snapshot =
+                 engine.newChangesSnapshot("test", fromSeqNo, toSeqNo, true, randomBoolean(), true, randomBoolean())) {
             IllegalStateException error = expectThrows(IllegalStateException.class, () -> drainAll(snapshot));
             assertThat(error.getMessage(),
                 containsString("Not all operations between from_seqno [" + fromSeqNo + "] and to_seqno [" + toSeqNo + "] found"));
         }
     }
+
+    public void testStats() throws Exception {
+        try (Store store = createStore();
+             Engine engine = createEngine(defaultSettings, store, createTempDir(), NoMergePolicy.INSTANCE)) {
+            final int numOps = between(100, 5000);
+            applyOperations(engine, generateHistoryOnReplica(numOps, randomBoolean(), randomBoolean(), randomBoolean()));
+            engine.refresh("test");
```

Comment:
Do we need this refresh? I would think the engine is responsible for ensuring a fresh enough searcher is available (calls `refreshIfNeeded()`)?